pkg/vpcmodel/subnetsConnectivity.go

```diff
@@ -252,14 +255,45 @@ func (v *VPCsubnetConnectivity) computeAllowedConnsCombined() error {
 			v.AllowedConnsCombined[src][dst] = combinedConns
 		}
 	}
+	return nil
+}
 
+func (v *VPCsubnetConnectivity) computeStatefulConnections() error {
+	for src, endpointConns := range v.AllowedConnsCombined {
+		for dst, conns := range endpointConns {
+			if conns.IsEmpty() {
+				continue
+			}
+			dstObj := v.cloudConfig.NameToResource[dst.Name()]
+			var otherDirectionConn *common.ConnectionSet = nil
+			switch dstObj.(type) {
+			case NodeSet:
+				otherDirectionConn = v.AllowedConnsCombined[dst][src]
+			case *ExternalNetwork:
+				// subnet to external node is stateful if the subnet's nacl allows egress from that node
+				otherDirectionConn = v.AllowedConns[src].EgressAllowedConns[dst]
+			default:
+			}
+			conns.IsStateful = common.StatefulFalse
+			if otherDirectionConn == nil {
+				continue
+			}
+			stateful, err := conns.ContainedIn(otherDirectionConn)
+			if err != nil {
+				return err
+			}
```

Comment:
why checking for a containment in specific direction? if the intersection between `conns` and `otherDirectionConn` is not empty, isn't it sufficient to be stateful?
Take a look at `nodesConnectivity.go` lines 237-247. Should also use `SwitchSrcDstPorts()` before checking the intersection here as well (as explained in https://github.com/np-guard/vpc-network-config-analyzer/issues/56#issuecomment-1586154619 ).

Reply:
There are two comments here: one about using `SwitchSrcDstPorts()` - sure, I'll take a look and use.
Not sure I understand the other comment. Does it refer to the following?
```
case *ExternalNetwork:
				// subnet to external node is stateful if the subnet's nacl allows egress from that node
				otherDirectionConn = v.AllowedConns[src].EgressAllowedConns[dst]
```

Reply:
I ended not adding `SwitchSrcDstPorts()` - since the ingress external nodes is missing here I think it is not really helpful here.

Reply:
why is `SwitchSrcDstPorts()` related to `ingress external nodes` ?
The connection set from `otherDirection` should have the ports flipped before intersection with `conns`.